deployment/terraform/iam.tf

```diff
@@ -0,0 +1,158 @@
+resource "aws_iam_instance_profile" "aws_lambda_profile" {
+  name = "aws-lambda-${var.application}-${var.environment}-profile"
+  role = aws_iam_role.aws_lambda_role.name
+  tags = {
+    git_commit           = "3a9544c19922bcd4ebcdb1f428df0532140a87fa"
+    git_file             = "deployment/terraform/iam.tf"
+    git_last_modified_at = "2022-11-01 18:05:43"
+    git_last_modified_by = "blackdenc@chop.edu"
+    git_modifiers        = "blackdenc"
+    git_org              = "kids-first"
+    git_repo             = "kf-lambda-member-update"
+    yor_trace            = "70ea0775-d288-4f95-9075-143df4e9fc1c"
+  }
+}
+
+resource "aws_iam_role" "aws_lambda_role" {
+  name = "aws-lambda-${var.application}-${var.environment}-role"
+
+  assume_role_policy = data.aws_iam_policy_document.lambda_assume_role_policy.json
+  tags = {
+    git_commit           = "3a9544c19922bcd4ebcdb1f428df0532140a87fa"
+    git_file             = "deployment/terraform/iam.tf"
+    git_last_modified_at = "2022-11-01 18:05:43"
+    git_last_modified_by = "blackdenc@chop.edu"
+    git_modifiers        = "blackdenc"
+    git_org              = "kids-first"
+    git_repo             = "kf-lambda-member-update"
+    yor_trace            = "4f9eca50-a260-4a24-ab05-8b343908e0d3"
+  }
+}
+
+data "aws_iam_policy_document" "lambda_assume_role_policy" {
+  statement {
+    principals {
+      type        = "Service"
+      identifiers = ["lambda.amazonaws.com"]
+    }
+    actions = [
+      "sts:AssumeRole"
+    ]
+  }
+}
+
+resource "aws_iam_policy" "aws_lambda_policy" {
+  name        = "aws-lambda-policy-${var.application}-${var.environment}"
+  description = "Policy for lambda"
+  policy      = data.aws_iam_policy_document.lambda_policy_document.json
+  tags = {
+    git_commit           = "3a9544c19922bcd4ebcdb1f428df0532140a87fa"
+    git_file             = "deployment/terraform/iam.tf"
+    git_last_modified_at = "2022-11-01 18:05:43"
+    git_last_modified_by = "blackdenc@chop.edu"
+    git_modifiers        = "blackdenc"
+    git_org              = "kids-first"
+    git_repo             = "kf-lambda-member-update"
+    yor_trace            = "e6357133-66ca-41e9-a8b9-119899f29c7e"
+  }
+}
+
+data "aws_iam_policy_document" "lambda_policy_document" {
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "s3:GetObject",
+      "s3:ListBucket",
+      "s3:GetBucketLocation",
+    ]
+
+    resources = [
+      "arn:aws:s3:::${var.payload_bucket}/*",
+      "arn:aws:s3:::${var.payload_bucket}",
```

Comment:
I wasn't sure if this should be left in or not for the function itself, or if this was only used for uploading the lambda zip in the previous configuration. I left it in out of caution, but we might be able to remove this and slim down this profile too.